media/js/base/mozilla-notification-banner.js

```diff
@@ -0,0 +1,269 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Create namespace
+if (typeof Mozilla === 'undefined') {
+    var Mozilla = {};
+}
+
+(function() {
+    'use strict';
+
+    var NotificationBanner = {};
+    var _options = {};
+
+    NotificationBanner.COOKIE_CODE_ID = 'mozilla-notification-banner';
+    NotificationBanner.COOKIE_EXPIRATION_DAYS = 21; // default cookie expiry 21 days
+    NotificationBanner.COOKIE_INTERACTION_VALUE = 'interacted';
+
+    /**
+     * Gets notification cookie if it exists.
+     * @return {String} cookie value if it exists else null.
+     */
+    NotificationBanner.getCookie = function() {
+        return Mozilla.Cookies.getItem(NotificationBanner.COOKIE_CODE_ID);
+    };
+
+    /**
+     * Generates cookie expiry data defaulting to 21 days.
+     * @return {Date}
+     */
+    NotificationBanner.cookieExpiresDate = function(date) {
+        var d = date || new Date();
+        d.setTime(d.getTime() + (NotificationBanner.COOKIE_EXPIRATION_DAYS * 24 * 60 * 60 * 1000));
+        return d.toUTCString();
+    };
+
+    /**
+     * Sets a cookie to register that notification has either been displayed or interacted with.
+     * @param {String} value - either the unique ID for a notification or COOKIE_INTERACTION_VALUE.
+     */
+    NotificationBanner.setCookie = function(value) {
+        Mozilla.Cookies.setItem(NotificationBanner.COOKIE_CODE_ID, value, NotificationBanner.cookieExpiresDate(), '/');
+    };
+
+    /**
+     * Redirects current page to CTA link target.
+     */
+    NotificationBanner.doRedirect = function(target) {
+        window.location.href = target;
+    };
+
+    /**
+     * Tracks interaction with notification close button in GA.
+     */
+    NotificationBanner.trackGAClose = function() {
+        window.dataLayer.push({
+            'data-banner-name': _options.name,
+            'data-banner-dismissal': '1',
+            'eAction' : 'banner dismissal',
+            'eLabel': _options.closeLabel,
+            'event': 'in-page-interaction'
+        });
+    };
+
+    /**
+     * Handles interaction with notification close button and removes from the DOM.
+     */
+    NotificationBanner.close = function() {
+        var notification = document.querySelector('.notification-banner');
+        var close = notification.querySelector('.notification-banner-close');
+
+        if (notification) {
+            close.removeEventListener('click', NotificationBanner.close, false);
```

Comment:
Should we unbind the `confirm` listener here as well? There's no way to get back to it once you close the notification, right?